coalib/bearlib/aspects/Formatting.py

```diff
@@ -0,0 +1,300 @@
+    """
+    class docs:
+        example = """
+        # There are two blank lines in this file, and the help improving
+        # on its readability by seperating imports from the rest of the code.
+        from coalib.bearlib.aspects import Root
+
+
+        @Root.subaspect
+        class SomeAspect:
+            class docs:
+                definition='this is an aspect'
+                ...
+        """
+        example_language = 'python'
+        imporatance_reason = """
+        Optional/useless blank lines affects the readability, its
+        maintainability and length.
+        """
+        fix_suggestions = """
+        Avoiding using usesless empty lines in your codebase.
+        """
+    blank_lines_after_declarations = Taste[int](
+        'Represents the number of blank lines after declarations',
+        (0, 1, 2), default=0)
+    blank_lines_after_blocks = Taste[int](
+        'Represents the number of blank lines after a block',
+        (0, 1, 2), default=1)
+    blank_lines_after_procedures = Taste[int](
+        'Represents the number of blank lines to use after a procedure or a'
+        ' function', (0, 1, 2), default=1)
+    blank_lines_after_class = Taste[int](
+        'Represents the number of blank lines to use after a class definition.',
+        (0, 1, 2, 3), default=2)
+    enforce_newline_at_EOF = Taste[bool](
+        'Checks if a file ends with a single newline.',
+        (True, False), default=True)
+
+
+@Spacing.subaspect
+class SpacesAroundOperator:
+    """
+    This aspect checks on the spacing around operators in your codebase.
+    """
+    class docs:
+        example = """
+        def f(a):
+            return 37+-+a[42-x :  y**3]  #Missing spaces around operators.
+        """
+        example_language = 'python'
+        importance_reason = """
+        Improves the readability and maintainability of your code.
+        """
+        fix_suggestions = """
+        """
+    spaces_around_operators = Taste[int](
+        'Represents the number of space to be used around operators.',
+        (0, 1), default=1)
+    spaces_before_colon = Taste[int](
+        'Represents the number of blank spaces before colons.',
+        (0, 1), default=0)
+    spaces_after_colon = Taste[int](
+        'Represents the number of blank spaces after colons.',
+        (0, 1), default=1)
+
+
+@Spacing.subaspect
+class Indentation:
+    """
+    This aspect checks on indentation in your file.
+    """
+    class docs:
+        example = """
+        # here is a example of code with some indentation issues
+
+        def func(a, b):
+        ...result = a**b  # Use 4 spaces per indentation level
+        \\treturn result  # Use either spaces or tabs for indentation
+        """
+        example_language = 'python'
+        importance_reason = """
+        Improves the readability of your code.
+        """
+        fix_suggestions = """
+        Use either spaces or tabs for indentation not both. In case you are
```

Comment:
the spaces vs tabs thing should be an own aspect and doesn't necessarily have anything to do with indentation IMO.

The indentation aspect will probably grow into a huge tree as soon as we start using it for more bears, maybe just remove it for now and make a comment? I feel like that'd be a months work on its own